specs/waku/waku.md

```diff
@@ -218,7 +218,7 @@ the same state.
 When a node is receiving other Waku messages from a peer before a Status
 message is received, the node MUST ignore these messages and SHOULD disconnect from that peer. Status messages received after the handshake is completed MUST also be ignored.
 
-The status message MUST contain an association list containing various options. All options within this association list are OPTIONAL, ordering of the key-value pairs is not guaranteed and therefore MUST NOT be relied on.
+The status message MUST contain an association list containing various options. All options within this association list are OPTIONAL, ordering of the key-value pairs is not guaranteed and therefore MUST NOT be relied on. Unknown keys in the association list MUST be ignored.
```

Comment:
I'd call this a SHOULD, there might be a ready for saving options for peers even if a node doesn't understand it at the time. Maybe?

Reply:
Rather: why should this be MUST vs SHOULD?

Marking it as as MUST also makes previous clients strictly incompatible, as opposed to this being a (very sane) recommendation.

Reply:
> I'd call this a SHOULD, there might be a ready for saving options for peers even if a node doesn't understand it at the time. Maybe?

I don't think we will ever save options of peers across the timeline of different versions.

> Marking it as as MUST also makes previous clients strictly incompatible, as opposed to this being a (very sane) recommendation.

I don't really understand the reasoning here. Why would they be "more" incompatible with MUST?

That being said, yeah, I was also doubting between MUST and SHOULD. I'll change it to SHOULD as it is rather recommendation of how older clients can try to remain compatible as much as possible. And it is consistent with the `Connecting to nodes with multiple versions` usage of SHOULD further down the document.

Reply:
The only reason I change it to `SHOULD` is because it would allow extensibility, people could add things here to make waku compatible subprotocols I guess.

Reply:
Having a MUST requirement in version n means that clients not fulfilling that MUST in version n-1 are incompatible. It is a "breaking changing" in that sense, because we are requiring more. As opposed to a strong recommendation.

Reply:
hmm, I think my text is being misunderstood. What I mean with unknown keys, is unknown to the receiver, because the receiver has an older spec implementation. Or because the sender does something custom, like your example @decanus. `MUST` will only make it sure that this works, as opposed to SHOULD, where an implementation could e.g. decide not to ignore it and  disconnect the peer (basically what status-go does now, and now Nimbus has no friends :( ).

But, I'll still change it to SHOULD, as it is a recommendation.

Reply:
@kdeme we could change it to SHOULD and add another recommendation that peers SHOULD not disconnect when seeing unrecognized fields, I agree that that would be bad practice.

Reply:
ah yes, true